### ISSUE TITLE
Blackarch slim installation failed

### ISSUE THREAD

**Issue by keremsrms0721** [BlackArch/blackarch-iso#53]
Type: Question

## Question
![blackarch](https://github.com/BlackArch/blackarch/assets/59603584/7d5a6ed6-ca2c-43c8-906d-5d64f030df3e)

I did not understand why this happens someone could help me about this error ? Because i installed the iso file again but this error happens again and again...

**Comment by noraj** [comment]
Could you explain the previous steps?

**Comment by anunna** [comment]
This can happen if there are issues connecting to mirrors, keyring issues or partial upgrades. Should  check if there is a good internet connection, using a close mirror,  and updating `archlinux-keyring` `blackarch-keyring`. 

But would be good to know also what you have done up to that point exactly.

**Comment by keremsrms0721** [comment]
> Could you explain the previous steps?

Hello noraj, I encountered an error while attempting to install BlackArch Linux Slim on VMware Workstation version 17.0.0. Initially, I suspected the error might be caused by the nameservers. Before installing, I changed the nameservers from 192.x.x.x to 8.8.8.8, 8.8.4.4, but the error persisted. I removed the network interface on VMware and attempted an offline installation, which succeeded. However, when I tried to run 'sudo pacman -Sy' for package installation, I encountered another error as shown in the attached screenshot. Could you please assist me in resolving this issue?
![image](https://github.com/BlackArch/blackarch-iso/assets/59603584/73e3a7e9-1646-4713-9288-ce68647e517e)

**Comment by noraj** [comment]
> This can happen if there are issues connecting to mirrors, keyring issues or partial upgrades. Should check if there is a good internet connection, using a close mirror, and updating `archlinux-keyring` `blackarch-keyring`.

As anunna said, that could be your mirror that seems to timeout. Try using another one.

**Comment by ScriptScorpion** [comment]
> > This can happen if there are issues connecting to mirrors, keyring issues or partial upgrades. Should check if there is a good internet connection, using a close mirror, and updating `archlinux-keyring` `blackarch-keyring`.
> 
> As anunna said, that could be your mirror that seems to timeout. Try using another one.
How can i using another one? If i cannot change mirror list because files are read-only in arch i can do it easily.

**Comment by noraj** [comment]
> How can i using another one?

Open a terminal and edit `/etc/pacman.d/blackarch-mirrorlist`.

> because files are read-only in arch

I don't know why you say that, but it's wrong.

**Comment by SRCGremlin** [comment]
```
#configure and test DNS
sudo nano /etc/pacman.conf
#remove community repos
sudo pacman -Sy
#confirm that databases sync effectively
sudo nano /etc/pacman.d/mirrorlist
sudo nano /etc/pacman.d/blackarch-mirrorlist
#confirm local servers are uncommented and therefore used
sudo pacman -S blackarch-config-calameres archlinux-keyring blackarch-keyring
#install potentially outdated packages and keyrings
```
#Stilll fails

#Root cause analysis 
Noted Issue in calamares session.log - installation is not respecting local changes to pacman repo files - regardless this link (mirrorspace) should still resolve. Calamares is also not respecting local DNS configurations. 

other locations this information is sourced from seem to include to fixed installation media:

/run/archiso/airootfs/etc (DNS hard fixed to 1.1.1.1 and 8.8.8.8)
/run/archiso/cowspace.....

potentially updating these to local sources before running the installer might work?

**Comment by ttzai** [comment]
so @noraj i tried use more than one verison of mirror but its still the same isusse

**Comment by noraj** [comment]
@anunna I guess now pacman could also fails because of https://archlinux.org/news/cleaning-up-old-repositories/
We should pack new ISOs

**Comment by anunna** [comment]
@noraj Yes, we have to get the ISOs updated. Otherwise if its not going to happen anytime soon we can place an updated `pacman.conf` file Calamares can access (e.g., PKGBUILD) and add a `shellprocess.conf` to the installer:

```
# shellprocess.conf
---
dontChroot: false
timeout: 30
script:
  - "cp /etc/calamares/pacman.conf ${ROOT}/etc/pacman.conf"
  - "chmod 644 ${ROOT}/etc/pacman.conf"
```

or probably a better way:

```
# shellprocess.conf
---
dontChroot: false
timeout: 30
script:
  - "cat > ${ROOT}/etc/pacman.conf << EOF
    [options]
    HoldPkg     = pacman glibc
    Architecture = auto
    ParallelDownloads = 5
    SigLevel    = Required DatabaseOptional
    LocalFileSigLevel = Optional
    
    [core]
    Include = /etc/pacman.d/mirrorlist
    
    [extra]
    Include = /etc/pacman.d/mirrorlist
    
    [community]
    Include = /etc/pacman.d/mirrorlist
    
    [BlackArch]
    Server = https://blackarch.whatever/\$repo/\$arch
    EOF"
  - "chmod 644 ${ROOT}/etc/pacman.conf"
  ```

I can get it updated

**Comment by viliamolo** [comment]
OK so the conclusion is, no solution, right? Just wait for update of ISO image, untill then you can go ....... How could happen that this ISO image is faulty and released? Because I didn´t find any workaround answer just unanswered question. What is nowadays happening? KDE developers have doesn´t care ongoing issues, Ubuntu cares only when Money can go out from it and every other distribution is "Half way ready" with tons of bugs and nogos. 

What is it? Why is it?

**Comment by noraj** [comment]
@viliamolo

- ISO are not mandatory, you can still configure BA repo over AL.
- Talking about KDE or Ubuntu is off topic here.
- Do you understand that FLOSS (free, libre and open-source software) is people working for you for free on their free time? It means we owe you nothing, you are not our customer, so please be respectful. If you find it too long to get a new ISO, you are free to work on the update yourself to offer it to the community. For Ubuntu, it's different since it backed by a company named Canonical that sells pro version and support. You can't compare a billion-dollar company vs 3 part-time unpaid volunteers.

**Comment by FershoUno** [comment]
Hello, the solution is to set the `update_db` parameter to `false` in the Calamares configuration.
This prevents the installer from attempting to update the repositories and allows you to continue uninstalling the packages listed in the Calamares configuration file.

Edit the file with:
sudo nano /etc/calamares/modules/packages.conf

Below is a screenshot showing how the configuration should look after applying the changes


<img width="709" height="552" alt="Image" src="https://github.com/user-attachments/assets/1aa9d09c-d82d-4854-b367-3b3d610bd1e6" />

**Comment by D3vil0p3r** [comment]
This issue has been fixed in https://github.com/BlackArch/blackarch-config-calamares/pull/5. Meanwhile, a workaround could be:

Boot into the BlackArch Slim live environment and run
```
sudo sed -i '/^\[community\]/,/^$/d' /etc/pacman.conf
sudo pacman -Syy
sudo pacman -S blackarch-config-calamares
```
Then launch Calamares and proceed with the installation. Once a new Slim ISO is released, this workaround should no longer be necessary.
